### PR TITLE
Compatibility with pandas 1

### PR DESCRIPTION
**Pull request by NicoRicardi**
Dear Oskar Weser,
I have:
1. a) removed the "by" from pd.sort_index(..., sort_remaining=by). One could check the pd version and based on that call the function differently, but I deem it superfluous as "by" was always left equal to "None" (at least as far as I saw).
b) This functions is called by "zmat.get_cartesian()" which is already tested, so no additional test is needed
2. some tests were failing because of leading spaces (e.g. " O 0.0 0.0 0.0" instead of "O 0.0 0.0 0.0"). I think this also depends on package versions and the test should accept both. I creat both versions and assert the result matches either.
3. I propose (feel free to discard) a cartesian.read_string() for users that may not be familiar with StringIO. 

Best, NR

### CODE REVIEW

**Comment by mcocdawc** [comment]
Thank you very much for this work and sorry for my late reply.
I completely agree with 1 and 2, but unfortunately have a different view on 3.

TLDR:
I will merge but immediately remove `read_string`, to have it in the history of the master branch.
I will update the documentation of `read_xyz` to make a reference to `io.StringIO`.

1. If a user does not know `io.StringIO`, they most probably don't know `Cartesian.read_string(...)` either. In any way they have to perform a Search and read through documentations.
Since `io.StringIO` is the idiomatic way to treat an in memory string as file and is part of the python Standard library, I think this is a good opportunity to learn it.
This will work then also with other libraries than `chemcoord`.
Since I like the UNIX philosophy of having small programs and libraries that do one thing well,
I don't want to add redundant pathways.

2. Nevertheless I completely agree with your implementation. If there really arises a need for a `read_string` method, it will be in the `master` history.

3. I will update the `read_xyz` documentation, to make it clearer, that `buf` can be a buffer or a filename.

**Comment by mcocdawc** [comment]
By the way:
The strange conversion bug came from a change in the numpy API.
https://github.com/numpy/numpy/issues/13666

I changed the relevant codelines to catch both `AttributeError` and `TypeError`.

**Comment by NicoRicardi** [comment]
I understand your point of view. It is your module, hence your choice.

I was personally not familiar with StringIO, but truly it is useful to learn it. I think it is good to have it as a function (but this can be done inside the script that import chemcoord) to avoid recalling the same StringIO (as if it were a standard string variable).
